sympy/combinatorics/rewritingsystem.py

```diff
@@ -330,14 +330,9 @@ def construct_automaton(self):
         The complete left hand side of the rules are the dead states of the automaton.
 
         '''
-        # Create a finite state machine as an instance of the StateMachine object
-        fsm = StateMachine('fsm')
-        fsm.add_state('start', state_type='s')
-
-        self.reduction_automaton = fsm
-        self.add_to_automaton(self.rules)
+        self._add_to_automaton(self.rules)
```

Comment:
If you do this, then you'll still have `_new_rules` contain all the rules. I think it might make sense to have a method `_update_automaton` that calls `_add_to_automaton(self._new_rules)` and then empties new rules by `_new_rules = {}`. It'd be a short method but you could use it here and also in the `reduce_using_automaton` method.

Reply:
By the time the control flow reaches here, the `_new_rules` dictionary will be empty since they are not yet computed as `self.reduction_automaton = None` by then.